### PR TITLE
Enable MySQL strict mode

### PR DESCRIPTION
**Pull request by eronisko**
⚠ This PR will likely unleash some unexpected bugs so we'll likely need to do some thorough testing before releasing

I've published this change to test: https://test.webumenia.sk/

* Enables MySQL strict mode
* Expands some columns to better accommodate real data
* Fixes tests to work in strict mode
* Minor change in AbstractHarvester for the `datestamp` field (cc @rastislav-chynoransky )

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: database/migrations/2017_05_03_115235_create_article_translations_table.php [hunk 1]
```diff
@@ -20,8 +20,8 @@ public function up()
 
             // translatable attributes
             $table->string('title')->default('');
-            $table->text('summary')->default('');
-            $table->text('content')->default('');
+            $table->text('summary');
+            $table->text('content');
```
Comment (igor-kamil) [inline database/migrations/2017_05_03_115235_create_article_translations_table.php:24]: shouldn't it be at least `->nullable()`?

**Review by igor-kamil — COMMENTED** [review]
right move. but ... I would be very "careful". this might break things way too easy... 
one of the issues is wrong default on every timestamp field `0000-00-00 00:00:00` what in compbination with `'strict' => true` prevents any alter on these tables
in my humble opinion - this should be fixed as well

File: database/migrations/2017_05_03_115235_create_article_translations_table.php (hunk 1 above)
Comment (eronisko) [reply database/migrations/2017_05_03_115235_create_article_translations_table.php:24]: This is actually changed to nullable in 2020_05_07_115534_allow_nulls_in_article_translations.php

**Comment by eronisko** [comment]
Deployed to https://test.webumenia.sk/ again

File: app/Http/Controllers/ItemController.php [hunk 2]
```diff
@@ -135,7 +135,9 @@ protected function processForm(Form $form) {
         $item = $form->getData();
         $form->handleRequest();
 
-        if ($form->isSubmitted() && $form->isValid()) {
+        if ($form->isSubmitted()) {
+            Validator::make(Request::all(), ['item.author' => 'required'])->validate();
```
Comment (igor-kamil) [inline app/Http/Controllers/ItemController.php:139]: 👌

**Review by igor-kamil — APPROVED** [review]
ready to go 🚀

**Pull request merged**